zerver/lib/export.py

```diff
@@ -992,7 +992,13 @@ def custom_fetch_user_profile(response: TableData, context: Context) -> None:
     realm = context["realm"]
     exportable_user_ids = context["exportable_user_ids"]
 
-    query = UserProfile.objects.filter(realm_id=realm.id)
+    query = UserProfile.objects.filter(realm_id=realm.id).exclude(
+        # These were, in some early versions of Zulip, inserted into
+        # the first realm that was created.  In those cases, rather
+        # than include them here, we will include them in the
+        # crossrealm user list, below.
+        email__in=[settings.WELCOME_BOT, settings.NOTIFICATION_BOT, settings.EMAIL_GATEWAY_BOT]
```

Comment:
Should we just use `settings.CROSS_REALM_BOT_EMAILS` here instead of a hard-coded list?

Reply:
We have some weird other existing hardcoding here:
https://github.com/zulip/zulip/blob/890732a88fd43d756025744d795b797fd27b0b8b/zerver/lib/export.py#L1025-L1042
...which I believe to be trying to manage folks who have `INTERNAL_BOT_DOMAIN` set to a non-default value somehow, which I also believe can't work because of things like `settings.WELCOME_BOT`.

I'll absolutely switch this use case, but we should add a task to do some cleanup on these email address settings.

Reply:
Filed #26295 as the follow-up.